### ISSUE TITLE
在 SecretFlow 中实现个性化联邦策略 FedBN

### ISSUE THREAD

**Issue by Candicepan** [secretflow/secretflow#994]
> 此 ISSUE 为 [隐语开源共建计划（SecretFlow Open Source Contribution Plan，简称 SF OSCP）](https://github.com/orgs/secretflow/discussions/1181)任务 ISSUE，欢迎社区开发者参与共建～
> 若有感兴趣想要认领的任务，但还未报名，辛苦先完成[报名](https://www.wjx.top/vm/QKio0dq.aspx# )进行哈～

## 任务介绍
- **任务名称**：在 SecretFlow 中实现个性化联邦策略 FedBN
- **技术方向**：FL/strategy
- **任务难度**：进阶🌟🌟
- **任务期望时间**：4-5周

## 详细要求
- 安全性：尽量少 reveal
- 功能性：在 SecretFlow 中实现论文[FedBN: Federated Learning on Non-IID Features via Local Batch Normalization](https://arxiv.org/abs/2102.07623)的算法。
- 正确性：至少复现论文中的一个实验验证实现的正确性
- 代码规范：Python 代码需要使用 black+isort 进行格式化（流水线包含代码规范检查卡点）
- 框架后端：实现 PyTorch 版本。若在完成 PyTorch 版本实现的基础上同步进行 TensorFlow 版本实现将额外累积一🌟。
- 提交说明：关联该 isuue 并提交代码至 https://github.com/secretflow/secretflow/blob/main/secretflow/ml/nn/fl/backend/torch/strategy/

## 能力要求
- 了解基本 git 操作
- 熟悉 深度学习、联邦学习
- 对 SecretFlow 有一定了解

## 操作说明
- 范例：[PyTorch 后端](https://github.com/secretflow/secretflow/blob/main/secretflow/ml/nn/fl/backend/torch/strategy/fed_prox.py) 和 [TensorFlow 后端](https://github.com/secretflow/secretflow/blob/main/secretflow/ml/nn/fl/backend/torch/strategy/fed_prox.py)

**Comment by elegaa** [comment]
elegaa Give it to me

**Comment by Candicepan** [comment]
> elegaa Give it to me

经沟通，该任务先进行回收～感谢您的认领呀 也欢迎下次继续参与隐语贡献～其他小伙伴可以再进行认领啦

**Comment by OkayYang** [comment]
OkayYang Give it to me

**Comment by Yeekin-GYJ** [comment]
> 好的杨给我

感谢认领，该任务完成过程中有什么问题可以在 issue 上直接反馈哈。也可以添加小助手微信：SecretFlow04 进行沟通～

**Comment by OkayYang** [comment]
你好，我看原论文中FedBN策略的定义只是在fedavg的聚合和更新的时候跳过了BN层，我看你们更新的最新代码中模型初始化的时候已经加入了skip_bn参数，那我定义FedBN的时候只需要将skip_bn默认为true好像就可以实现效果。

**Comment by OkayYang** [comment]
> > 好的杨给我
> 
> 
> 
> 感谢认领，该任务完成过程中有什么问题可以在 issue 上直接反馈哈。也可以添加小助手微信：SecretFlow04 进行沟通～

你好，我看原论文中FedBN策略的定义只是在fedavg的聚合和更新的时候跳过了BN层，我看你们更新的最新代码中模型初始化的时候已经加入了skip_bn参数，那我定义FedBN的时候只需要将skip_bn默认为true好像就可以实现效果。
